### PR TITLE
adding path as an attribute on the folder types

### PR DESCRIPTION
**Pull request by gitsome**

### CODE REVIEW

File: vfile-reporter-folder-json.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 const { sep } = require("path");
 
 class vFolder {
-  constructor({ name = "", children = [] } = {}) {
+  constructor({ name = "", children = [], parentPath = "" } = {}) {
     return {
       type: "folder",
+      path: parentPath + name + "/",
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.js]: consider using `path.resolve` or `path.join` to ensure cross platform support on windows.

File: vfile-reporter-folder-json.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const { sep } = require("path");
 
 class vFolder {
-  constructor({ name = "", children = [] } = {}) {
+  constructor({ name = "", children = [], parentPath = "" } = {}) {
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.js]: Could `parentPath` be considered a working directory in this case?
perhaps `cwd` as the variable name?

File: vfile-reporter-folder-json.spec.js [hunk 3]
```diff
@@ -55,6 +55,18 @@ test("two sub-folders generate", () => {
...
+test("folders have path attribute", () => {
+  const file = vfile({
+    path: join("example", "README.md"),
+    contents: "# Hello"
+  });
+  const folder = reporter([file], { raw: true });
+  expect(folder).toHaveProperty("path", "/");
+
+  const subFolder = folder.children.find(vfile => vfile.type === "folder");
+  expect(subFolder).toHaveProperty("path", "/example/");
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.spec.js]: on windows `\example\`, see `path.sep`

File: vfile-reporter-folder-json.spec.js [hunk 4]
```diff
@@ -55,6 +55,18 @@ test("two sub-folders generate", () => {
   expect(folder).toHaveProperty("children.1.children.0", file2);
 });
 
+test("folders have path attribute", () => {
+  const file = vfile({
+    path: join("example", "README.md"),
+    contents: "# Hello"
+  });
+  const folder = reporter([file], { raw: true });
+  expect(folder).toHaveProperty("path", "/");
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.spec.js]: on windows `\`.
See `path.sep`

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]
Thanks @gitsome! :bow:
Having additional metadata on folders will be valuable. :+1:

File: vfile-reporter-folder-json.spec.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 const vfile = require("vfile");
 const reporter = require("./vfile-reporter-folder-json");
-const { join } = require("path");
+const path = require("path");
+const join = path.join;
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.spec.js]: :art: `const {join} = require("path");`

File: vfile-reporter-folder-json.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
-const { sep } = require("path");
+const path = require("path");
+const sep = path.sep;
```
Comment (ChristianMurphy) [inline vfile-reporter-folder-json.js]: :art: `const {join, sep} = require("path");`

**Review by ChristianMurphy — APPROVED** [review]
Thanks @gitsome!
Two small stylistic requests then this should be good to go!

File: vfile-reporter-folder-json.js (hunk 6 above)
Comment (gitsome) [reply vfile-reporter-folder-json.js]: cool

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**